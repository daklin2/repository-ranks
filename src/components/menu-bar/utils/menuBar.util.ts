import {Languages} from "../../../models/repositoryModel";

export const listOfLanguages = [
  {
    displayName: "JavaScript",
    queryName: Languages.JAVA_SCRIPT,
  },
  {
    displayName: "TypeScript",
    queryName: Languages.TYPE_SCRIPT,
  },
  {
    displayName: "Java",
    queryName: Languages.JAVA,
  },
  {
    displayName: "Python",
    queryName: Languages.PYTHON,
  },
  {
    displayName: "Ruby",
    queryName: Languages.RUBY,
  },
  {
    displayName: "Scala",
    queryName: Languages.SCALA,
  },
  {
    displayName: "All",
    queryName: Languages.ALL,
  }
];