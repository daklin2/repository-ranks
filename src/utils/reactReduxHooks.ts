import { TypedUseSelectorHook, useDispatch as useDispatchOrg, useSelector as useSelectorOrg } from 'react-redux';
import type { AppState, AppDispatch } from '../store/store';

export const useDispatch = () => useDispatchOrg<AppDispatch>();
export const useSelector: TypedUseSelectorHook<AppState> = useSelectorOrg;