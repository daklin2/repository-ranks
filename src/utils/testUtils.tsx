import {Provider} from "react-redux";
import {Store} from "@reduxjs/toolkit";
import {mount} from "enzyme";

export const mountWrapper = (store: Store, element: JSX.Element) => {
  return mount(
    <Provider store={store}>
      {element}
    </Provider>
  )
}