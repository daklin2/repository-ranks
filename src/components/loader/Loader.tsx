import "./Loader.style.scss";
import {useSelector} from "../../utils/reactReduxHooks";

const Loader = () => {
  const isLoading = useSelector((state) => state.loader.isLoading)

  return isLoading ? (
    <div className={"loader__background"}>
      <div className="lds-roller">
        <div/>
        <div/>
        <div/>
        <div/>
        <div/>
        <div/>
        <div/>
        <div/>
      </div>
    </div>
  ) : null
}

export default Loader;
