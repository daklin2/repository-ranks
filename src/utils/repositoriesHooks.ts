import {changeSelectedLanguage, getLanguageStats} from "../store/languageMenuSlice";
import {useDispatch} from "./reactReduxHooks";
import {Languages} from "../models/repositoryModel";

export const useChangeSelectedLanguage = () => {
  const dispatch = useDispatch();

  return (language: Languages) => {
    dispatch(getLanguageStats(language))
      .then(() => {
        dispatch(changeSelectedLanguage(language))
      })
  }

}