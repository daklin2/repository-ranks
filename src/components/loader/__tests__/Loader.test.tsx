import {mount, shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Loader from "../Loader";
import * as ReactReduxHooks from "../../../utils/reactReduxHooks";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const useDispatchSpy = jest.spyOn(ReactReduxHooks, "useDispatch");
useDispatchSpy.mockImplementation(() => defaultStore.dispatch)

const defaultStoreState = {
  loader: {
    isLoading: true,
  }
};
const defaultStore = configureStore([thunk])(defaultStoreState);

describe("Loader", () => {
  const useSelectorSpy = jest.spyOn(ReactReduxHooks, "useSelector");

  beforeEach(() => {
    useSelectorSpy.mockImplementation((state) => state(defaultStore.getState() as any));
  });
  afterEach(() => {
    jest.clearAllMocks();
    defaultStore.clearActions();
  });

  it("should render Loader", () => {
    expect(toJson(shallow(<Loader/>))).toMatchSnapshot();
  });

  it("shouldn't render Loader if nothing loading", () => {
    const editedStoreState = {
      loader: {
        isLoading: false,
      }
    }
    const store = configureStore([thunk])(editedStoreState);
    useSelectorSpy.mockImplementation((state) => state(store.getState() as any));
    const wrapper = mount(<Loader />);

    expect(wrapper.find("div").length).toBe(0);
  });
})