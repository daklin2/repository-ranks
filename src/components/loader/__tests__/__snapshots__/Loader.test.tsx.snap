// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render Loader 1`] = `
<div
  className="loader__background"
>
  <div
    className="lds-roller"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
